packages/react-instantsearch/src/connectors/connectInfiniteHits.js

```diff
@@ -66,7 +66,7 @@ export default createConnector({
     const id = getId();
     const nextPage = searchState[id] ?
       Number(searchState[id]) + 1 :
-      1;
+      0;
```

Comment:
Looks legit based on Algolia API knowledge, but we've had discussion and we decided to fix the Algolia API and provide a user understandable numeration. If there is a bug, it lies where the correction is done not here.